File: src/frontend/web/study-room-system/src/lib/types/Seat.ts
import type { Room } from "./Room";
import type { Booking } from "./Booking";

export interface Seat {
    id : string;
    row : number;
    col : number;
    room : Room | null;
    bookings?: Booking[];
}

export interface SeatState {
    id: string | null;
    row: number;
    col: number;
    open: boolean;     // 是否创建
    booked: boolean;   // 是否被预约
}

File: src/backend/cs/StudyRoomSystem/StudyRoomSystem.Core/Structs/Entity/Seat.cs
namespace StudyRoomSystem.Core.Structs.Entity;

public record Seat
{
    public Guid Id { get; set; } = Ulid.NewUlid().ToGuid();
    public required Guid RoomId { get; set; }
    public required int Row { get; set; }
    public required int Col { get; set; }

    public virtual Room Room { get; set; } = null!;

    public virtual ICollection<Booking> Bookings { get; set; } = [];
}